File: ehr/test/src/org/labkey/test/components/ehr/panel/SingleAnimalFilterType.java
package org.labkey.test.components.ehr.panel;

import org.labkey.test.components.ldk.panel.SingleSubjectFilterType;
import org.labkey.test.pages.ehr.AnimalHistoryPage;

public class SingleAnimalFilterType<A extends AnimalHistoryPage> extends SingleSubjectFilterType<A>
{

    public SingleAnimalFilterType(A sourcePage)
    {
        super(sourcePage);
    }

    @Override
    protected String getIdPrefix()
    {
        return "ehr-singleanimalfiltertype";
    }
}

File: ehr/resources/web/ehr/panel/SingleAnimalFilterType.js

Ext4.define('EHR.panel.SingleAnimalFilterType', {
    extend: 'LDK.panel.SingleSubjectFilterType',
    alias: 'widget.ehr-singleanimalfiltertype',

    statics: {
        filterName: 'singleSubject',
        DEFAULT_LABEL: 'Single Animal'
    },

    validateAlias: function(tab) {
        if (!this.isValid()) {
            tab.items[0].report.warning = 'Animal ID not found';
        }
        else {
            tab.items[0].report.warning = undefined;
        }

        return tab;
    },

    getAlias: function (subjectArray, callback, panel, tab, forceRefresh) {
        this.aliasTableConfig(subjectArray);

        this.aliasTable.success = function (results) {
            this.handleAliases(results);
            this.getSubjectMessages();
            this.validateAlias(tab);
            callback.call(panel, this.handleFilters(tab, this.subjects), forceRefresh);
        };

        LABKEY.Query.selectRows(this.aliasTable);
    }

});
